Sources/SPTDataLoader.m

```diff
@@ -252,6 +314,14 @@ - (void)requestIsWaitingForConnectivity:(SPTDataLoaderRequest *)request
         [self executeDelegateBlock:^{
             [self.delegate dataLoader:self requestIsWaitingForConnectivity:request];
         }];
+    } else {
+        [self executeDelegateBlock:^{
+            void (^handler)(SPTDataLoaderResult *) = [self.requestHandlers objectForKey:request];
+            handler([SPTDataLoaderResult fromRequest:request kind:SPTDataLoaderResultIsWaitingForConnectivity]);
+            @synchronized(self.requestHandlers) {
+                [self.requestHandlers removeObjectForKey:request];
+            }
```

Comment:
Do you really want to remove the handler here when waiting for connectivity? Then nothing will fire when the requests succeeds or fails

Reply:
@dmakarov is this comment still valid? I cannot see the handler being removed

Reply:
The handler is no longer removed.  I removed the removing of the handler in this case.